File: src/main/scala/arimitsu/sf/cql/v3/messages/Authenticate.scala
package arimitsu.sf.cql.v3.messages

import java.nio.ByteBuffer
import arimitsu.sf.cql.v3.util.Notation

/**
 * Created by sxend on 14/06/07.
 */
case class Authenticate(className:String)

object AuthenticateParser extends ResponseParser[Authenticate]{
  override def parse(body: ByteBuffer): Authenticate = {
    Authenticate(Notation.getString(body))
  }
}

File: src/main/scala/arimitsu/sf/cql/v3/messages/Authenticate.java
package arimitsu.sf.cql.v3.messages;

import arimitsu.sf.cql.v3.Parser;
import arimitsu.sf.cql.v3.util.Notation;

import java.nio.ByteBuffer;

/**
 * Created by sxend on 14/06/07.
 */
public class Authenticate {
    public final String className;

    public Authenticate(String className) {
        this.className = className;
    }

    public static final Parser<Authenticate> AuthenticateParser = new Parser<Authenticate>() {
        @Override
        public Authenticate parse(ByteBuffer body) {
            return new Authenticate(Notation.getString(body));
        }
    };
}
